OMCompiler/SimulationRuntime/c/meta/realString.cpp

```diff
@@ -89,11 +89,11 @@ modelica_string _ryu_realString(modelica_real r)
 // the minimal decimal or exponential version of it to stdout
 void _ryu_to_minimal(const char *str1, char *buf) {
   int sign;  // Sign of the number
-  char digits[32];  // Digits of the mantissa
+  char digits[32] = {0};  // Digits of the mantissa
   int ndec; // Number of digits after decimal point in the input
   int exp;  // Exponent in the input
-  char str2[32]; // Stores the decimal output
-  char *ptr1 = str1; // Movable pointer to the input
+  char str2[32] = {0}; // Stores the decimal output
+  char *ptr1 = (char*)str1; // Movable pointer to the input
```

Comment:
This should be:
```
const char *ptr1 = str1;
```
Actually, `ptr1` is not even needed since it's just an alias for `str1`, you can just use `str1` in the code instead and skip `ptr1`.

Reply:
When scanning the input, I need to move the pointer with `ptr1++` or `++ptr1`. I guess I can't do that with `str1`, can I?

Reply:
You can, a function parameter is just a local variable inside the function.

Reply:
OK, as long as I don't need to rewind it as I do with `ptr`. But I like it best to keep ptr1 and ptr2 for simmetry 😃

Reply:
I'll fix that

Reply:
I also thought it was not necessary to initialize str2, as the the code is expected to write the decimal string and terminate it with a null character. But I agree it's safer to initialize it to zeros, just in case. This also removes the need to terminate str2, as it will be terminated automatically. I'll do that.